turbinia/workers/binary_extractor.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Task to extract binary files from an evidence object provided."""
+
+from __future__ import unicode_literals
+
+import os
+
+from turbinia import config
+from turbinia.workers import TurbiniaTask
+from turbinia.evidence import BinaryExtraction
+
+
+class BinaryExtractorTask(TurbiniaTask):
+  """Extract binaries out of evidence and provide JSON file with hashes."""
+
+  def run(self, evidence, result):
+    """Task that extracts binaries with image_export.py.
+
+    Args:
+        evidence: Path to data to process.
+        result: A TurbiniaTaskResult object to place task results into.
+
+    Returns:
+        TurbiniaTaskResult object.
+    """
+
+    config.LoadConfig()
+    binary_extraction_evidence = BinaryExtraction()
+
+    binary_extraction_evidence.local_path = self.output_dir
```

Comment:
Is your intention to save all the binaries as well?  Or just the log file?  Right now without adding some extra processing this directory of binary files won't get saved by itself. It will still be in the output directory, which in a local installation is the same, but for cloud instances it won't copy everything.   There is actually a new Evidence type coming very soon that could work well for this though.  Look at the `CompressedDirectory` evidence type in this PR: https://github.com/google/turbinia/pull/442/files.   You could potentially extend that in this PR, and then you'd have a tarball of all binaries from the system as output.

Reply:
Yes, the idea here is to extract the binaries as well, along with json file with all the hashes. Using CompressedDirectory sounds like a good idea here, I changed my to code to use it. Is there something I need to add on my side to make Turbinia copy the output evidence to cloud bucket?

Reply:
I believe you don't need to make any other changes.  As long as the Evidence property for `copyable` is True, then it should automatically copy it to GCS as long as there is a GCS path set in the config, and as long as the Evidence is attached to the result object properly (which should happen automatically if you're using `execute()` as you are).